torch/utils/_sympy/solve.py

```diff
@@ -128,6 +132,26 @@ def _try_isolate_lhs(
             assert op is not None
             e = op(lhs, rhs)
 
+    ################################################################################
+    # left-hand side is CleanDiv
+    ################################################################################
+    if (
+        isinstance(e, sympy.Eq)
+        and isinstance(e.lhs, CleanDiv)
+        and e.lhs.is_integer
+        and e.rhs.is_integer
+        and e.lhs.has(thing)
+    ):
+        numer, denom = e.lhs.args
+        thing_in_numer = numer.has(thing)
+        thing_in_denom = denom.has(thing)
+
+        # only support thing in denominator for now
+        if thing_in_denom and thing_in_numer != thing_in_denom:
+            return _try_isolate_lhs(
+                sympy.Eq(denom, CleanDiv(numer, e.rhs)), thing, floordiv_inequality
```

Comment:
Potential infinite loop when rhs contains thing.

```
CleanDiv(8, G(thing)) = F(thing)
```

you then simplify to 

```
G(thing) = CleanDiv(8, F(thing))
```

but this will trigger this logic again.

Do you still have enough juice if you exclude thing from rhs?